kasa/smart/smartdevice.py

```diff
@@ -790,3 +790,42 @@ def _get_device_type_from_components(
             return DeviceType.Thermostat
         _LOGGER.warning("Unknown device type, falling back to plug")
         return DeviceType.Plug
+
+    @staticmethod
+    def _get_device_info(
+        info: dict[str, Any], discovery_info: dict[str, Any] | None
+    ) -> _DeviceInfo:
+        """Get model information for a device."""
+        di = info["get_device_info"]
+        components = [comp["id"] for comp in info["component_nego"]["component_list"]]
+        short_name = di["model"]
+        region = None
+        if discovery_info:
```

Comment:
Please separate logical elements by adding some newlines in-between. It is really hard to read & follow the flow when reading this from top to bottom.

Reply:
Done